### PR TITLE
Base Docker image off LSIO Beets image

### PR DESCRIPTION
**Pull request by montchr**
I've been running my fork of Betanin based off the `linuxserver/beets` Docker image on my personal media server. I love Betanin (nice job, thank you!), though I needed to make some customizations for it to work with my setup.

I've aimed for flexibility, rather than tailor these changes to my specific needs, but I'm sure I've made some oversights! I'm new-ish to Docker, and have zero experience developing Python applications. So I would consider this PR a work-in-progress, but would like to at least put it out there as a draft.

If you have the time, I'd appreciate any feedback or suggestions, and please feel free to tweak as needed.

## Summary of Changes

- Base the Docker image off `linuxserver/beets`, which provides Beets dependencies for all of its plugins. This change should address #22.
- Remove `--copy` flag from the `beet import` command to allow customization within `config.yml`. Also represented in #33.
- Change home directory to `/config`, which is used by the LSIO image's `abc` user.
- Set `BEETS_DIR` to Beets' XDG config path.
- Set default `GID` to `1001`. This might be specific to my setup, so I'd be fine with reverting that if desired.
- Fix Docker image build issues
- ⚠️ Forces usage of bleeding-edge Beets version by removing the `1.4.9` version constraint and installing Beets `HEAD` version via git URL. This fixes fatal errors with later versions of Python. Forcing this isn't ideal since it's potentially unstable, so I should probably add a tag in the Dockerfile to allow customizing the LSIO tag specified in the Dockerfile (see TODOs below).

## TODO

- [ ] Add `Dockerfile` tag to choose stable or nightly beets image/version.
- [ ] Avoid installing Beets in `Dockerfile` – use the version installed by the LSIO image.
- [ ] Add `Dockerfile` tag to replace `Dockerfile.dev`
- [ ] Ensure local development of Betanin can continue while retaining these changes
- [ ] Look for ways of reducing the size of the image if possible
- [ ] Can Betanin co-exist with the Beets web UI plugin?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
+
+# Clean up
+RUN apk del --purge \
+        build-deps
+RUN rm -rf /tmp/*
+
+VOLUME \
+    "/config/.config/beets" \
+    "/config/.config/betanin" \
+    "/config/.local/share/betanin" \
```
Comment (sentriz) [inline Dockerfile:33]: unfortunately I think changing the betanin/beets data paths would be a breaking change for users of the container
eg. I personally have in my compose file
```
volumes:
- ./data:/root/.local/share/betanin/
- ./config:/root/.config/betanin/
- ./beets:/root/.config/beets/
```

though I do agree `/config` is a better place for this stuff.
maybe some symlinks could help for some backwards compatibility?

File: Dockerfile.dev [hunk 2]
```diff
@@ -4,18 +4,16 @@ FROM python:3.6.6-alpine3.6
 LABEL maintainer="Senan Kelly <senan@senan.xyz>"
 RUN apk add --no-cache libev build-base libffi-dev
 COPY ./requirements-docker.txt /
-RUN --mount=type=cache,target=/root/.cache/pip \
```
Comment (sentriz) [inline Dockerfile.dev:7]: any reason to get rid of the `--mount=type=cache,target=/root/.cache/pip`? when running with `COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1` this seems to help with faster builds while development (though maybe not I haven't done comparisons)

File: betanin/paths.py [hunk 3]
```diff
@@ -15,7 +15,8 @@ def _first_existing(paths):
 # dir
 DATA_DIR = xdg.BaseDirectory.save_data_path("betanin")
 CONFIG_DIR = xdg.BaseDirectory.save_config_path("betanin")
-BEETS_DIR = os.path.expanduser("~/.config/beets/")
+BEETS_DIR = xdg.BaseDirectory.save_config_path("beets")
```
Comment (sentriz) [inline betanin/paths.py:18]: 👌

**Review by sentriz — COMMENTED** [review]
thanks! this is really cool. i like it :) i'll be happy to test it when you think its ready

left some feedback/questions

**Comment by edgar-vincent** [comment]
Any news on this excellent feature?

**Comment by montchr** [comment]
> Any news on this excellent feature?

I haven't had much chance to work on this PR lately, but I haven't forgotten about it. I'm hoping to revisit in the near future.

**Comment by sentriz** [comment]
hello! thanks very much for this but closing for now. please see #46

**Pull request closed without merging**